proj/src/main/webapp/chat.js

```diff
@@ -754,6 +769,24 @@ async function addUsernameToMessage(uid, preview) {
     });
 }
 
+function pfpOnInput() {
+    const input = document.getElementById("pfp-upload");
+    const pfp = input.files[0];
+
+    const currentUid = firebase.auth().currentUser.uid;
+    const pfpStorageRef = firebase.storage().ref('/'+currentUid+'/pfp.png');
+    pfpStorageRef.put(pfp);
+
+    pfpStorageRef.getDownloadURL().then(function(url) {
+        const userPfpRef = firebase.database().ref('/users/'+currentUid+'/photo');
+        userPfpRef.set(pfpStorageRef.toString());
+
+        userPfp = document.getElementById('user-pfp');
+        userPfp.src = url;
+    })
+    input.files = null; // clear the input
+}
+
```

Comment:
I'm not sure what this method is doing. Is this for users to upload their own pictures? Can you add some comments to say what this is doing?

Are you storing the images at `/{uid}/pfp.png` would it make sense to put it one layer in something like `profilePictures/{uid}/pfp.png` in case you want to add other things to the storage?